internal/worker/removal/worker.go

```diff
@@ -62,8 +85,48 @@ func NewWorker(config Config) (worker.Worker, error) {
 }
 
 func (w *removalWorker) loop() (err error) {
-	<-w.catacomb.Dying()
-	return w.catacomb.Err()
+	watch, err := w.cfg.RemovalService.WatchRemovals()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	if err := w.catacomb.Add(watch); err != nil {
+		return errors.Capture(err)
+	}
+
+	ctx := w.catacomb.Context(context.Background())
+
+	timer := w.cfg.Clock.NewTimer(jobCheckMaxInterval)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-w.catacomb.Dying():
+			return w.catacomb.ErrDying()
+		case jobIDs := <-watch.Changes():
+			w.cfg.Logger.Infof(ctx, "got removal job changes: %v", jobIDs)
+			if err := w.processRemovalJobs(); err != nil {
+				return errors.Capture(err)
```

Comment:
We'll have to be careful here, as a generic error can crash the worker. I've been bitten by this before.